### PR TITLE
Add taxonomy app

### PR DESCRIPTION
**Pull request by engineervix**
This PR adds a new taxonomy app to the project, with a taxonomy modeladmin menu item and categories sub-menu item

### CODE REVIEW

File: wagtailio/taxonomy/wagtail_hooks.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from wagtailio.taxonomy.models import Category
+
+
+class CategoryModelAdmin(ModelAdmin):
+    model = Category
+    menu_icon = "tag"
+
+
+class TaxonomiesModelAdminGroup(ModelAdminGroup):
+    menu_label = "Taxonomies"
```
Comment (zerolab) [inline wagtailio/taxonomy/wagtail_hooks.py:16]: nitpick: 
```suggestion
    menu_label = "Taxonomy"
```

File: wagtailio/taxonomy/tests.py [hunk 2]
```diff
@@ -0,0 +1 @@
+# Create your tests here.
```
Comment (William-Blackie) [inline wagtailio/taxonomy/tests.py:1]: Suggestion: If we're not using them, do we need this files here?

File: wagtailio/taxonomy/wagtail_hooks.py (hunk 1 above)
Comment (engineervix) [reply wagtailio/taxonomy/wagtail_hooks.py]: Thanks, updated in 74d3e24

File: wagtailio/taxonomy/tests.py (hunk 2 above)
Comment (engineervix) [reply wagtailio/taxonomy/tests.py]: Thanks, removed in 70deda8

**Pull request merged**